### PR TITLE
Gomod

### LINKED ISSUE #524: Switch to Go modules
**What feature would you like to be added?**

With Go modules being more and more stable and becoming default soon, we should consider switching to Go modules. Kubernetes has also switched to Go modules as of 1.15.

AC

* [x] Switch to Go modules
* [x] Make code generation works after switching to Go modules

Resources

https://github.com/kubernetes/kubernetes/pull/74877
[Introduction to Go modules by Jordan Liggitt](https://www.youtube.com/watch?v=OrMA16ASz2U)

### PR DESCRIPTION
**Pull request by eqrx**
**What this PR does / why we need it**:
Makes KubeOne a go module

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #524 

**Special notes for your reviewer**:
None

**Does this PR introduce a user-facing change?**:
```release-note
KubeOne is now a go module
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -45,18 +45,10 @@ lint:
 test:
 	CGO_ENABLED=1 go test -race ./...
 
-.PHONY: dep
-dep:
-	dep ensure -v
```
Comment (kron4eg) [inline Makefile]: I think we can replace this with `go mod download`

File: Makefile [hunk 2]
```diff
@@ -45,18 +45,10 @@ lint:
...
-
 .PHONY: licence-check
 licence-check:
 	wwhrd check
 
-.PHONY: check-dependencies
-check-dependencies:
-	dep check
```
Comment (kron4eg) [inline Makefile]: and this with `go mod verify`

File: Makefile (hunk 1 above)
Comment (eqrx) [reply Makefile]: Should we still call that target "dep"?

Comment (kron4eg) [reply Makefile]: yeah, SGTM, since it's from "dependency" so whatever 🤷‍♂️

**Comment by eqrx** [comment]
Currently not possible because of the kubernetes code-generator. See issue for more details.

**Pull request merged**